craft.lic

```diff
@@ -0,0 +1,291 @@
+    end
+  end
+
+  def train_forging
+    eligible = @settings.train_workorders & %w[Blacksmithing Weaponsmithing]
+    unless eligible.empty?
+      return unless money_for_training?(5000, 'Forging')
+      wait_for_script_to_complete('workorders', [eligible.first])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    rank = DRSkill.getrank('Forging')
+    if rank <= 25 # Tier 1 - Extremely Easy
+      smith('a shallow metal cup')
+    elsif rank <= 50 # Tier 2 - Very Easy
+      smith('a short metal mug')
+    elsif rank <= 100 # Tier 3 - Easy
+      smith('a back scratcher')
+    elsif rank <= 175 # Tier 4 - Simple
+      smith('a metal ankle band')
+    elsif rank <= 300 # Tier 5 - Basic
+      smith('a metal lockpick ring')
+    elsif rank <= 425 # Tier 6 - Somewhat Challenging
+      smith('a metal armband')
+    elsif rank <= 550 # Tier 7 - Challenging
+      smith('some metal clippers')
+    elsif rank <= 700 # Tier 8 - Complicated
+      smith('some squat knitting needles')
+      # Journeyman book: triangular wire sieve, bent <metal> scissors, knobby sewing needles, squat knitting needles, narrow <metal> awl
+    elsif rank <= 850 # Tier 9 - Intricate
+      smith('serrated hide scraper')
+      # Master book: beveled wood shaper, serrated hide scraper, compact <metal> awl, compact <metal> awl, round pestle
+    elsif rank <= 1175 # Tier 10 - Difficult
+      smith('serrated scissors')
+      # Master book: slender <metal> awl, serrated scissors, grooved pestle, oblong wire sieve
+    elsif rank <= 1400 # Tier 11 - Very Difficult
+      smith('thin sewing needles')
+      # Master book: jagged wood shaper, thin sewing needles
+    else # Tier 12 - Extremely Difficult
+      smith('trapezoidal wire sieve')
+      # Master book: trapezoidal wire sieve
+    end
+  end
+
+  def train_alchemy
+    if @settings.train_workorders.include?('Remedies')
+      return unless money_for_training?(5000, 'Alchemy')
+      wait_for_script_to_complete('workorders', ['Remedies'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    beep
+    echo('*** NOT YET IMPLEMENTED ***')
+    @settings.crossing_training.delete('Alchemy')
+  end
+
+  def money_for_training?(amount, skill)
+    if ensure_copper_on_hand(amount, @settings)
+      true
+    else
+      echo("Low on funds, removing #{skill} from training")
+      @settings.crossing_training.delete(skill)
+      false
+    end
+  end
+
+  def sew(chapter, unique_name, item)
+    check_yarn
+
+    walk_to(@training_room)
+    check_listening
+    check_teaching
+
+    wait_for_script_to_complete('sew', ['trash', 'knitting', chapter, unique_name, item, 'wool'])
+
+    bput('stow my yarn', 'You put', 'Stow what')
+  end
+
+  def shape(chapter, unique_name, item)
+    check_wood
+
+    find_shaping_room(@hometown, @training_room)
+    check_listening
+    check_teaching
```

Comment:
❓ What happens if you listen and try to teach?

Reply:
I'm assuming its assumed someone wouldn't have both classes_to_teach and listen: true in their yaml at the same time, this was lifted straight out of current crossing-training.

Reply:
Ok